### DISCUSSION TITLE
Unable to control ESP32 via web UI if not on same wifi network also unable to connect to MQTT

### DISCUSSION THREAD

**Discussion by Rolls76** [arendst/Tasmota#17719] [Q&A]
Hi All, 

Newbie here, I am trying to set up an ESP32 for the first time & having difficulties.  

I followed the setup procedure step for step on a youtube video to using the web installer & got as far as the WEB UI.  As I said it works fine on the same wifi network.  If I turn the WIFI off on my phone & try to control the unit through my mobile 4G connection it wont respond.  (ESP device is still connected to my WIFI) 

I am also unable to connect to the MQTT broker & keep getting the following error message:
  
10:40:59.029 MQT: Attempting connection...
10:41:02.095 MQT: Connect failed to test.mosquitto.org:1883, rc -5. Retry in 10 sec

I am using MQTT lens & the mosquitto.test broker just to get started with learning.  

I have a spare router on hand & have tried connecting the ESP to that one with the same results.

Any tips would be greatly appreciated please

**Comment by barbudor** [comment]
Your devices are connected to your home network which is private, behind your ISP router
When turning wifi off, your phone is connected to your cell phone operator network
By default there is no routing between those networks so it is normal that any of your internal IP address such as 192.168.1.X is not reachable from outside. It would be a HUGE security breach other wise

Concerning connection to test.mosquitto.org, the error codes are documented here : https://tasmota.github.io/docs/MQTT/#return-codes-rc
-5 is MQTT_DNS_DISCONNECTED
Which means that your Tasmota doesn't have access to a name resolver. Which means you haven't configured the network correctly
- If using DHCP, then your ISP router should have provided you the info
- If using static IP? you may have configured the IP address of Tasmota but not configured the gateway nor the DNS

Check the [IpAddress](https://tasmota.github.io/docs/Commands/#ipaddress) command to set up the missing bits
